### PR TITLE
Updates TimeSeriesBindaryDao to include missing version date on retrieve

### PR DESCRIPTION
**Pull request by rma-bryson**
Adds in timezone to binary and text TS retrieve, and date version type to text TS retrieval.

### CODE REVIEW

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java [hunk 1]
```diff
@@ -454,4 +458,37 @@ protected static <R> R connectionResult(DSLContext dslContext, ConnectionCallabl
...
+
+        } else {
+            dateVersionType = VersionType.SINGLE_VERSION;
+        }
+
+        return dateVersionType;
+    }
+
+    private boolean isVersioned(String names, String office) {
+        return connectionResult(dsl, connection -> {
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: open an issue about caching this in memory.

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: also this should be in timeseries dao not this low level

File: cwms-data-api/src/main/java/cwms/cda/data/dao/binarytimeseries/TimeSeriesBinaryDao.java [hunk 2]
```diff
@@ -157,43 +156,21 @@ public BinaryTimeSeries retrieve(String officeId, String tsId, String mask,
...
                 .withOfficeId(officeId)
                 .withName(tsId)
                 .withBinaryValues(binRows)
                 .withDateVersionType(versionType)
+                .withVersionDate(versionInstant)
+                .withTimeZone(timeZoneId)
                 .build();
     }
 
-    @NotNull
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/binarytimeseries/TimeSeriesBinaryDao.java:169]: why was this moved?

File: cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/TimeSeriesTextDao.java [hunk 3]
```diff
@@ -45,29 +45,6 @@ public TextTimeSeries retrieveFromDao(@NotNull String officeId, @NotNull String
                 .build();
     }
 
-    @NotNull
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/texttimeseries/TimeSeriesTextDao.java:48]: now i see, reducing dupes, belongs in time series dao

**Review by MikeNeilson — CHANGES_REQUESTED** [review]

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java [hunk 4]
```diff
@@ -454,4 +458,37 @@ protected static <R> R connectionResult(DSLContext dslContext, ConnectionCallabl
...
+            Configuration configuration = getDslContext(connection, office).configuration();
+            return OracleTypeMap.parseBool(CWMS_TS_PACKAGE.call_IS_TSID_VERSIONED(configuration,
+                    names, office));
+        });
+    }
+
```
Comment (rma-rripken) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: These all seem pretty version specific.  The JooqDao class doesn't feel like the right spot for them.

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: yeah eithet TimseriesDao or a new Ts Utility class.

**Comment by rma-rripken** [comment]
Related to #547.   Can the offset be set also?

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java (hunk 4 above)
Comment (rma-bryson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: Used by TimeseriesDao, TimeSeriesBinaryDao, and RegularTimeSeriesTextDao currently. Since there's no common base class there, utility class seems like a good candidate?

Comment (MikeNeilson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: Yeah. but I think also fine to just stick in TimesSeriesDAO. it's time series operation regardless of the underlying data storage.

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java (hunk 1 above)
Comment (rma-bryson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: moved into TimeSeriesDaoImpl

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java [hunk 5]
```diff
@@ -24,6 +24,8 @@
 
 package cwms.cda.data.dao;
 
+import cwms.cda.api.enums.VersionType;
```
Comment (MikeNeilson) [inline cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: VersionType should be unused now. Check for other unused.

**Review by MikeNeilson — CHANGES_REQUESTED** [review]
Organization look good now. Sanity check the what should be unused imports.

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java (hunk 5 above)
Comment (rma-bryson) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: Removed unused imports

File: cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java (hunk 1 above)
Comment (rma-rripken) [reply cwms-data-api/src/main/java/cwms/cda/data/dao/JooqDao.java]: https://github.com/USACE/cwms-data-api/issues/607

**Comment by rma-rripken** [comment]
Mike, what changes do you still want to see?

**Review by adamkorynta — APPROVED** [review]

**Review by MikeNeilson — APPROVED** [review]

**Comment by MikeNeilson** [comment]
> Mike, what changes do you still want to see?

None, just didn't realize my initial request didn't get cleared after the changes.
